### ISSUE TITLE
Leaderboard calculations

### ISSUE THREAD

**Issue by jdaudier** [loicginoux/joanne_project#261]
When I add up the points on http://www.foodrubix.com/admin/users/chrissy, I get 50.

But the leaderboard has Chrissy at 43.

**Comment by jdaudier** [comment]
Is this the same bug? I have negative points again.
![joanne](https://f.cloud.github.com/assets/1666031/130656/cf510116-702d-11e2-8832-9b45299ca162.png)

**Comment by jdaudier** [comment]
And Vicky has negative points too.  Do you know what is going on?
![vicky](https://f.cloud.github.com/assets/1666031/130660/fc7fc672-702d-11e2-99d9-b16ae60f8ed1.png)

**Comment by loicginoux** [comment]
Ok as a quick fix, I recalculated the points for everyone, so now it's temporarily fixed.
This happened when you deleted the user yesterday.

explanation of what happened:
deleting a user delete all his photos (with their comments and likes), all his comments, all his likes and all his friendship.
these deletions means that points must also be removed for certain people (for example if you delete a user and this user had a comment on a photo of the user "X", then "X" must loos 2 points because the comment has also been deleted)

there is two issues with that:
- the first one is my responsabilities, I just remove points in both total leaderboard and monthly leaderboard, although I should have removed points from the monthly leaderboard only for the iteim deleted from this month.
- second one, deleting a user and all his comments/likes/etc... will make other people loose points. These people won't understand why they suddenly got their points deleted.

So several questions:
- should we remove points to people who interacted with the deleted user? ie if I commented on his photo that get deleted, then my comment will be deleted and I will loose points... but is that fair?

let me know, if it's not clear, and what you think about this...
